src/lib/seam/connect/routes/access-codes.ts

```diff
@@ -0,0 +1,226 @@
+
+export class SeamHttpAccessCodes {
+  client: Axios
+
+  constructor(apiKeyOrOptions: string | SeamHttpOptions) {
+    const options = parseOptions(apiKeyOrOptions)
+    this.client = createAxiosClient(options)
+  }
+
+  static fromClient(
+    client: SeamHttpOptionsWithClient['client'],
+    options: Omit<SeamHttpOptionsWithClient, 'client'> = {},
+  ): SeamHttpAccessCodes {
+    const opts = { ...options, client }
+    if (!isSeamHttpOptionsWithClient(opts)) {
+      throw new SeamHttpInvalidOptionsError('Missing client')
+    }
+    return new SeamHttpAccessCodes(opts)
+  }
+
+  static fromApiKey(
+    apiKey: SeamHttpOptionsWithApiKey['apiKey'],
+    options: Omit<SeamHttpOptionsWithApiKey, 'apiKey'> = {},
+  ): SeamHttpAccessCodes {
+    const opts = { ...options, apiKey }
+    if (!isSeamHttpOptionsWithApiKey(opts)) {
+      throw new SeamHttpInvalidOptionsError('Missing apiKey')
+    }
+    return new SeamHttpAccessCodes(opts)
+  }
+
+  static fromClientSessionToken(
+    clientSessionToken: SeamHttpOptionsWithClientSessionToken['clientSessionToken'],
+    options: Omit<
+      SeamHttpOptionsWithClientSessionToken,
+      'clientSessionToken'
+    > = {},
+  ): SeamHttpAccessCodes {
+    const opts = { ...options, clientSessionToken }
+    if (!isSeamHttpOptionsWithClientSessionToken(opts)) {
+      throw new SeamHttpInvalidOptionsError('Missing clientSessionToken')
+    }
+    return new SeamHttpAccessCodes(opts)
+  }
+
+  get unmanaged(): SeamHttpAccessCodesUnmanaged {
+    return SeamHttpAccessCodesUnmanaged.fromClient(this.client)
+  }
+
+  async create(
+    body: AccessCodesCreateBody,
+  ): Promise<AccessCodesCreateResponse['access_code']> {
+    const { data } = await this.client.request<AccessCodesCreateResponse>({
+      url: '/access_codes/create',
+      method: 'post',
+      data: body,
+    })
+    return data.access_code
+  }
+
+  async createMultiple(
+    body: AccessCodesCreateMultipleBody,
+  ): Promise<AccessCodesCreateMultipleResponse['access_codes']> {
+    const { data } =
+      await this.client.request<AccessCodesCreateMultipleResponse>({
+        url: '/access_codes/create_multiple',
+        method: 'post',
+        data: body,
+      })
+    return data.access_codes
+  }
+
+  async delete(body: AccessCodesDeleteBody): Promise<void> {
+    await this.client.request<AccessCodesDeleteResponse>({
+      url: '/access_codes/delete',
+      method: 'post',
+      data: body,
+    })
+  }
+
+  async generateCode(
+    params?: AccessCodesGenerateCodeParams,
+  ): Promise<AccessCodesGenerateCodeResponse['generated_code']> {
+    const { data } = await this.client.request<AccessCodesGenerateCodeResponse>(
+      {
+        url: '/access_codes/generate_code',
+        method: 'get',
+        params,
+      },
+    )
+    return data.generated_code
+  }
+
+  async get(
+    body: AccessCodesGetBody,
+  ): Promise<AccessCodesGetResponse['access_code']> {
```

Comment:
If the public types are complete, it might be interesting to compare the DX of this return type with 
```ts
  ): Promise<AccessCode[]> {
```